pages/articles/docker.md

```diff
@@ -0,0 +1,201 @@
+        Start-Process msiexec.exe -Wait -ArgumentList "/I $((Get-Location).Path)\wsl_update_x64.msi /quiet"
+        ```
+
+    2. Reboot the machine before you continue
+
+    3. Finish installing WSL:
+
+        ```powershell
+        wsl --set-default-version 2
+        wsl --install --distribution Ubuntu
+        ```
+
+    4. Install docker in WSL:
+
+        ```bash
+        bash <(curl -s https://get.docker.com/)
+        sudo usermod -aG docker $USER
+        newgrp docker
+        ```
+
+- Windows:
+
+    ```powershell
+    iex ((New-Object System.Net.WebClient).DownloadString('https://chocolatey.org/install.ps1'))
+    choco install -y docker-engine
+    ```
+
+## Using PnP.PowerShell
+
+Sometimes you want to run inline PnP.PowerShell commands.
+
+- Linux/WSL/Mac OS:
+
+    \<to be verified\>
+
+    ```bash
+    docker run --rm -it m365pnp/powershell
+    ```
+
+- Windows (run in PowerShell console):
+
+    ```powershell
+    docker run --rm -it m365pnp/powershell:1.10.0-nanoserver-1809
+    ```
+
+After that you can start running commands like `Connect-PnPOnline`.
+
+Mind you that in the case above, the container will have an isolated disk system so the commands that you run inside the container will not be able to access files from your host machine. However, sometimes you might want to run a script or use some files from your host OS (for example, when you have a ps1 file in a git repository that you cloned to your laptop). In this case you will need to mount a directory on your host as a volume inside the container. See the following examples.
+
+- Linux/WSL/Mac OS:
+
+    \<to be verified\>
+
+    ```bash
+    docker run --rm -it -v "$(pwd):/home" -w /home m365pnp/powershell
+    ```
+
+- Windows (run in PowerShell console):
+
+    ```powershell
+    docker run --rm -it -v "${pwd}:C:\workplace" -w C:\workplace m365pnp/powershell:1.10.0-nanoserver-1809
+    ```
+
+In such container you can run `Get-ChildItem` and see the contents of the current directory.
+
+The examples above start new container and allow you to interactively use container, running inline commands. In other cases however, you might want to just start a container for a short time (a few seconds maybe) and run a ps1 script unattededly and then automatically terminate the container. Here is how you can do it.
+
+- Linux/WSL/Mac OS:
+
+    \<to be verified\>
+
+    ```bash
+    docker run --rm -v "$(pwd):/home" -w /home m365pnp/powershell pwsh test.ps1
+    ```
+
+- Windows (run in PowerShell console):
+
+    ```powershell
+    docker run --rm -v "${pwd}:C:\workplace" -w C:\workplace m365pnp/powershell:1.10.0-nanoserver-1809 pwsh test.ps1
+    ```
+
+Finally, your scripts might have parametrization so that you can run the same code in different cases/environments. This is the way to provide variable values.
+
+- Linux/WSL/Mac OS:
+
+    \<to be verified\>
+
+    ```bash
+    ParameterValue="test"
```

Comment:
here as well, can we please remove this section, the `to be verified` one ? If we are not sure, lets not add it and confuse users